src/Microsoft.AspNetCore.Razor.Evolution/RazorInstrumentationIRPass.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Globalization;
+using Microsoft.AspNetCore.Razor.Evolution.Intermediate;
+using Microsoft.AspNetCore.Razor.Evolution.Legacy;
+using System.Diagnostics;
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    internal class RazorInstrumentationIRPass : RazorIRPassBase
+    {
+        public override int Order => 175;
+
+        public override DocumentIRNode ExecuteCore(DocumentIRNode irDocument)
+        {
+            var walker = new InstrumentationIRNodeWalker();
+            walker.VisitDocument(irDocument);
+
+            return irDocument;
+        }
+
+        internal class InstrumentationIRNodeWalker : RazorIRNodeWalker
+        {
+            private bool _skipNext = false;
```

Comment:
This is just here to prevent this code from going in to an infinite loop. I agree this needs some sort of comment.